### DISCUSSION TITLE
5.4.0: improvement of `imageLayer` markers & add `videoLayer`

### DISCUSSION THREAD

**Discussion by mistic100** [mistic100/Photo-Sphere-Viewer#1088] [Announcements]
The markers plugin `imageLayer` can now be positionned by providing the coordinates of the four corners of the image, allowing to match a panel/screen/billboard in the panorama. There is also a new `videoLayer` marker for videos. [Check the demo](https://photo-sphere-viewer.js.org/demos/markers/layers.html).

**Comment by haldo98** [comment]
I am trying the videoLayer but I do not understand the layer positioning:

I tried the "position (one value) + size"  but it seems that the position is related to the center of the video and not the upper left corner as I thought.

is it correct?

**Reply by mistic100** [reply]
"position" + "size" works exactly like other type of markers, by taking "anchor" into account (default to "center center")

or you can use the new system by providing four coordinates to "position" and no "size" or "anchor", this is what is used in the demo

**Comment by haldo98** [comment]
yep... I forgot the anchor....

Thank You....

no sound support?

**Reply by mistic100** [reply]
most (if not all) browsers prevents videos to automatically start if they are not muted, so it is muted by default

One can get the video element and unmute it from button click for example (it needs to be an user action somehow, it cannot be automatic or the browser will prevents it)

`markersPlugin.getMarker('my-video').video.muted = false`

**Comment by haldo98** [comment]
I added a videoLayer but when I switch the Hide/show marker button on the navbar  the video remains always visible.

Is this the desired behaviour?

**Reply by mistic100** [reply]
Please create a ticket
